clusterous/cluster.py

```diff
@@ -1233,26 +1246,24 @@ def terminate_cluster(self, leave_shared_volume, force_delete_shared_volume):
             self._terminate_instances_and_wait(conn, instances)
 
         # Delete shared volume
-        volumes = conn.get_all_volumes(filters={'tag:Attached':self.cluster_name})
-        if volumes:
-            byo_volume = True if volumes else False
+        if shared_volume:
             if byo_volume:
-                shared_volume = volumes[0]
-                shared_volume.remove_tags({'Attached': self.cluster_name})
-            else:
-                volumes = conn.get_all_volumes(filters={'tag:{0}'.format(defaults.instance_tag_key):self.cluster_name})
-                shared_volume = volumes[0]
-    
-            if leave_shared_volume:
-                self._logger.info('Shared volume "{0}" has not been deleted'.format(shared_volume.id))
-            else:
-                if force_delete_shared_volume or not byo_volume:
+                if force_delete_shared_volume:
                     if shared_volume.delete():
                         self._logger.info('Shared volume "{0}" has been deleted'.format(shared_volume.id))
                     else:
                         self._logger.error('Unable to delete volume in {0}: {1}'.format(self.cluster_name, shared_volume.id))
                 else:
-                    self._logger.info('Shared volume "{0}" has not been deleted'.format(shared_volume.id))
+                    shared_volume.remove_tags({'Attached': self.cluster_name})
+                    self._logger.info('Leaving shared volume "{0}"'.format(shared_volume.id))
+            else:
+                if leave_shared_volume:
+                    self._logger.info('Leaving shared volume "{0}"'.format(shared_volume.id))
+                else:
+                    if shared_volume.delete():
+                        self._logger.info('Shared volume "{0}" has been deleted'.format(shared_volume.id))
```

Comment:
Probably shouldn't print the shared volume id as default behaviour, as it's inconsistent with the other messages. Perhaps make it debug() or just show "Deleting shared volume", and separately show the shared volume ID with debug().